packages/vanilla/src/types.ts

```diff
@@ -31,19 +51,45 @@ export type ParallelApiSuccessCallback = (response: ParallelApiRecord) => void
 export type ParallelApiErrorCallback = (reason: any) => void
 
 type SubscribeEvents = 'auth.login' | 'auth.logout' | 'auth.statusChange' | 'auth.authResponseChange'
-type EventHandler = (response: any) => void
+type OAuthErrorCode =
+  | 'invalid_request'
+  | 'invalid_client'
+  | 'invalid_grant'
+  | 'unauthorized_client'
+  | 'unsupported_grant_type'
+type SubscriptionEvent = {
+  status: 'not_authorized' | 'unknown' | 'connected'
+  error?: OAuthErrorCode
+  errorDescription?: string
+  authResponse?: {
+    expires_in: number
+    token_type: 'bearer'
+    access_token: string
+    refresh_token: string
+    refresh_expires_in: number
+  }
+}
+type SubscriptionHandler = (response: SubscriptionEvent) => void
+
+export interface LoginOptions {
+  email?: string
+  first_name?: string
+  last_name?: string
+  expected_entity_id?: string
+  expected_entity_type?: 'self' | BusinessType
```

Comment:
Think we need two more here:
```suggestion
  expected_entity_type?: 'self' | BusinessType
  expected_business_name?: string
  partner_supporting?: string
```